src/pantry.js

```diff
@@ -1,7 +1,97 @@
 class Pantry {
   constructor(userIngredients) {
     this.contents = userIngredients;
+    this.filtered
+    this.goodAmount;
+    this.notEnough = []
+  }
+
+  determineIfUserHasEnoughIngredients(recipe) {
+    let AllIngredientInPantryIds = this.contents.map(ing =>{
+      return ing.ingredient 
+    })
+
+     this.filtered = recipe.ingredients.filter(ingredient => {
+      return AllIngredientInPantryIds.includes(ingredient.id)
+    })
+
+     this.goodAmount =  this.filtered.filter(ing =>{
+      if (AllIngredientInPantryIds.indexOf(ing.id) > -1) {
+        if (!this.contents[AllIngredientInPantryIds.indexOf(ing.id)].amount >= ing.quantity.amount) {
+          this.notEnough.push(ing)
+        }
+        return this.contents[AllIngredientInPantryIds.indexOf(ing.id)].amount >= ing.quantity.amount
+      }
+    })
+    return this.goodAmount.length === this.filtered.length
+  }
+
+  findAmountMissing(recipe) {
+//invoke determineIfUser... ()recipe
+    //if returns false => 
+    //find which ingredients are not satisfied by pantry
+  if (!this.determineIfUserHasEnoughIngredients(recipe)) { 
+    return this.notEnough.length
```

Comment:
what should be done with `this.notEnough` next?  Should we be pushing ingredient IDs to this initially empty array? (once we've isolated the data based on items the user still needs for a recipe)

Reply:
I feel that we should return the exact ingredient and amount missing but I'm not sure if we should make a separate function for that or refactor what we have to return it here.

Reply:
this can probably be done within the function we already have but if it violates SRP, we can certainly separate with a helper function!